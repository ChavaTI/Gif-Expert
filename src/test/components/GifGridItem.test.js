import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('GifGridItem tests', () => {
  let title;
  let url;
  let wrapper;
  beforeEach(() => {
    title = 'Test';
    url = 'test.com';
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });
  it('should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have a <p> with the title', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe(title);
  });
  it('should have the correct image', () => {
    const img = wrapper.find('img');
    expect( img.props().src ).toBe(url);
    expect( img.props().alt ).toBe(title);
  });
  it('should be a card', () => {
    const root = wrapper.find('div');
    expect(root.props().className.includes('card')).toBe(true);
  });
});
