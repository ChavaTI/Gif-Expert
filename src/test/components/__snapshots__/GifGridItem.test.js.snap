// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests should show the component correctly 1`] = `
<div
  className="card"
>
  <img
    alt="Test"
    src="test.com"
  />
  <p>
     
    Test
     
  </p>
</div>
`;
